lighthouse-core/runner.js

```diff
@@ -186,14 +183,20 @@ class Runner {
           const artifactError = artifacts[artifactName];
           log.warn('Runner', `${artifactName} gatherer, required by audit ${audit.meta.name},` +
             ` encountered an error: ${artifactError.message}`);
-          return audit.generateAuditResult({
-            rawValue: -1,
-            debugString: artifactError.message
-          });
+          throw new Error(
+              `Required ${artifactName} gatherer encountered an error: ${artifactError.message}`);
         }
       }
 
       return audit.audit(artifacts);
+    }).catch(err => {
+      if (err.fatal) {
+        throw err;
+      }
+
+      // Non-fatal errors become the result of the audit. Decorate with name.
+      err.name = audit.meta.name;
```

Comment:
> hmm this will throw away TypeError and soforth right? Do we want to preserve any of that?

yeah, I wasn't wild about this, but the assumption that the audit result will have the audit name on `name` is built deep in several places in the runner and if you provide `auditResults` in the config (which affects a bunch of tests), so this seemed easier. Not sure if we really care about the name.

> What does it look like to the user, similar to debugString?

somehow I skipped this pretty important step :)